test/minikube/minikube_base_database_test.go

```diff
@@ -369,11 +380,102 @@ func TestKubernetesBackupDatabase(t *testing.T) {
 		populateCreateDBData(t, namespaceName, admin0)
 
 		defer testlib.Teardown(testlib.TEARDOWN_BACKUP)
-		backupDatabase(t, namespaceName, admin0, "demo", &databaseOptions)
+		backupDatabase(t, namespaceName, admin0, "demo", false, &databaseOptions)
+	})
+
+	t.Run("nobackupSMBackup", func(t *testing.T) {
+		t.Skip("See: https://github.com/nuodb/nuodb-helm-charts/issues/64")
+		defer testlib.Teardown(testlib.TEARDOWN_DATABASE)
+		databaseOptions := helm.Options{
+			SetValues: map[string]string{
+				"database.sm.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+				"database.sm.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+				"database.te.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+				"database.te.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+				"backup.persistence.enabled":            "true",
+				"backup.persistence.size":               "1Gi",
+				"database.sm.HotCopy.replicas":          "0",
+				"database.sm.HotCopy.enabled":           "true",
+				"database.sm.noHotCopy.replicas":        "1",
+			},
+		}
+
+		databaseChartName := testlib.StartDatabase(t, namespaceName, admin0, &databaseOptions)
+		opts := k8s.NewKubectlOptions("", "")
+		opts.Namespace = namespaceName
+		opt := testlib.GetExtractedOptions(&databaseOptions)
+		smPodName := fmt.Sprintf("sm-%s-nuodb-%s-%s-hotcopy-0", databaseChartName, opt.ClusterName, opt.DbName)
+		fmt.Printf(k8s.RunKubectlAndGetOutputE(t, opts, "label", "--overwrite", "pods", "-n", namespaceName, smPodName, "role=nohotcopy"))
+		populateCreateDBData(t, namespaceName, admin0)
+		defer testlib.Teardown(testlib.TEARDOWN_BACKUP)
+		backupDatabase(t, namespaceName, admin0, "demo", false, &databaseOptions)
 	})
+
+
+	t.Run("backupNoSMs", func(t *testing.T) {
```

Comment:
this test will be flaky, since the backup could be taken before you scale down the SM (which in turn will kill the TE). The better option is to start the database with 1 non-hotcopy SM and 0 hotcopy SMs.

Reply:
Good idea, will do.

Reply:
In this case it becomes exactly the same as the above test where I removed the labeling and is skipped (https://github.com/nuodb/nuodb-helm-charts/issues/64). I will remove this test.